### PR TITLE
Do not write subscribe to discovery on auto-subscribe after follow

### PR DESCRIPTION
**Pull request by michellebrier**
### Description
Moved the subscribe on follow functionality to the indexer in discovery when indexing FOLLOW events [here](https://github.com/AudiusProject/audius-protocol/pull/4592). The indexer will validate the follow before creating both the follow and subscribe records. This means we don't need to issue a separate SUBSCRIBE event after the FOLLOW is confirmed in the client, as the time required to wait for the sequential relays to complete was affecting subscriptions writes.

Add an onFollow flag when processing the SET_NOTIFICATION_SUBSCRIPTION action and don't write to discovery when it's true. Right now we are still dual writing to both discovery and identity so the identity write completes as usual. We'll switch over to discovery after this and the backend change are deployed and the discovery subscriptions table has been backfilled.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?
Tested in web with breakpoints. Verified in DN that the indexer handled the auto subscribe/unsubscribe on follow/unfollow

### How will this change be monitored?
Will query subscriptions tables in identity / discovery to ensure parity before switching over and deprecating identity subscriptions.

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/web/src/pages/profile-page/ProfilePageProvider.tsx [hunk 1]
```diff
@@ -238,7 +238,6 @@ class ProfilePage extends PureComponent<ProfilePageProps, ProfilePageState> {
     if (!profile) return
     const userId = profile.user_id
     this.props.onUnfollow(userId)
-    this.props.setNotificationSubscription(userId, false)
```
Comment (michellebrier) [inline packages/web/src/pages/profile-page/ProfilePageProvider.tsx:241]: unfollow already dispatches a `SET_NOTIFICATION_SUBSCRIPTION` action. this was causing 2 unsubscribe requests to be issued to identity

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mbrier/plat-572/client_subscribe_on_follow_em

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mbrier/plat-572/client_subscribe_on_follow_em

**Review by sliptype — APPROVED** [review]
Looks good to me! Will the notification bell still be displayed as enabled immediately after following?

**Comment by michellebrier** [comment]
> Looks good to me! Will the notification bell still be displayed as enabled immediately after following?

Yes!

**Pull request merged**